protocol/synthetix/contracts/storage/AccountDelegationIntents.sol

```diff
@@ -0,0 +1,170 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.11 <0.9.0;
+
+import "./DelegationIntent.sol";
+import "@synthetixio/core-contracts/contracts/utils/SetUtil.sol";
+import "@synthetixio/core-contracts/contracts/utils/SafeCast.sol";
+import "../interfaces/IVaultModule.sol";
+import "./Account.sol";
+
+/**
+ * @title Represents a delegation (or undelegation) intent.
+ */
+library AccountDelegationIntents {
+    using SafeCastI256 for int256;
+    using SafeCastU128 for uint128;
+    using SafeCastU256 for uint256;
+    using SetUtil for SetUtil.UintSet;
+    using SetUtil for SetUtil.AddressSet;
+    using DelegationIntent for DelegationIntent.Data;
+    using Account for Account.Data;
+
+    struct Data {
+        uint128 accountId;
+        uint128 delegationIntentsEpoch; // nonce used to nuke previous intents using a new era (useful on liquidations)
+        SetUtil.UintSet intentsId;
+        mapping(bytes32 => SetUtil.UintSet) intentsByPair; // poolId/collateralType => intentIds[]
+        // accounting for the intents collateral delegated
+        // Per Collateral
+        SetUtil.AddressSet delegatedCollaterals;
+        mapping(address => int256) netDelegatedAmountPerCollateral; // collateralType => net delegatedCollateralAmount
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id.
+     */
+    function load(
```

Comment:
if we got rid of this load function and instead embedded this data structure into the `Account`, it would reduce load complexity and heiarchally store data in a convenient manner